crates/build/src/lib.rs

```diff
@@ -686,29 +720,40 @@ pub fn execute(args: ExecuteArgs) -> Result<BuildResult> {
                 verbosity,
                 " {} {}",
                 format!("{build_steps}").bold(),
-                "Post processing wasm file".bright_green().bold()
+                "Post processing code".bright_green().bold()
             );
             build_steps.increment_current();
-            post_process_wasm(&crate_metadata, skip_wasm_validation, &verbosity)?;
 
-            maybe_println!(
-                verbosity,
-                " {} {}",
-                format!("{build_steps}").bold(),
-                "Optimizing wasm file".bright_green().bold()
-            );
-            build_steps.increment_current();
+            let original_size =
+                fs::metadata(&crate_metadata.original_code)?.len() as f64 / 1000.0;
+
+            match target {
+                Target::Wasm => {
+                    post_process_wasm(&crate_metadata, skip_wasm_validation, &verbosity)?;
+                    let handler =
+                        WasmOptHandler::new(optimization_passes, keep_debug_symbols)?;
+                    handler.optimize(
+                        &crate_metadata.dest_code,
+                        &crate_metadata.contract_artifact_name,
+                    )?;
+                }
+                Target::RiscV => {
+                    fs::copy(&crate_metadata.original_code, &crate_metadata.dest_code)?;
+                }
+            }
```

Comment:
A downside of having both targets produce the final `contract.code` file is that you can't distinguish from the file extension the format of the contents i.e. whether it contains `wasm` or `risc-v` code.

Reply:
I did this on purpose because I don't want stale build artifacts for other architectures lingering around in the directory. Then you still don't really now what is in your bundle without looking at timestamps. It looked to me as the more confusing choice.

Reply:
I think the best solution is to keep the arch specific extension and just clean up old files on rebuild.

Reply:
The alternative is to have the artifacts in target specific dirs, there already exists e.g. `target/ink/wasm32-unknown-unknown`

![image](https://user-images.githubusercontent.com/75586/235889559-4a0527ef-a227-481c-b7c0-d89ebb23e3ce.png)

In any case we should add a field to the metadata which specifies the target code architecture, maybe somewhere in here
![image](https://user-images.githubusercontent.com/75586/235890845-1da3b933-d2e8-45db-8cb6-c4e910956a5b.png)

Reply:
Yes that might work to. But I went with keeping them in the top level dir and just clean up stale files. Metadata changes will be done in a follow up PR.